### DISCUSSION TITLE
Is there any way to disable a rule?

### DISCUSSION THREAD

**Discussion by denver1111** [hyperjumptech/grule-rule-engine#312] [Q&A] [answered]
Let's say I have 10 rules already written in my .grl file, after some time I want to disable 3 of them.
So that when I call engine.Execute() it should evaluate against enabled 7 rules.
I don't want to remove 3 rules, I want to keep it as disabled.

**Comment by denver1111** [comment]
As per Drools, there is an attribute 'enabled' using which we can control enable/disable a rule.
Looks like 'enabled' attribute is not implemented in Grule yet.

**Comment by newm4n** [comment, marked as answer]
You could create a rule with highest salience and immediately retract rules that you don't need.

```
rule RunMeFirst "Disable rules we donot want to be evaluated" salience 99999 {
    when
         true
    then
        retract("doNotRunRule1");
        retract("doNotRunRule2");
        retract("doNotRunRule3");
        retract("doNotRunRule4");
        retract("RunMeFirst");
}
```
